lib/system.nim

```diff
@@ -2431,13 +2433,13 @@ when not defined(nimscript):
       ##
       ## The allocated memory belongs to its allocating thread!
       ## Use `createShared <#createShared,typedesc>`_ to allocate from a shared heap.
-      cast[ptr T](alloc0(sizeof(T) * size))
+      cast[ptr T](alloc0(sizeof(T) * size, alignOf(T)))
 
-    proc realloc*(p: pointer, newSize: Natural): pointer {.noconv, rtl, tags: [],
+    proc realloc*(p: pointer, newSize: Natural, align = MemAlign): pointer {.noconv, rtl, tags: [],
                                                            benign, raises: [].}
       ## Grows or shrinks a given memory block.
       ##
-      ## If `p` is **nil** then a new memory block is returned.
+      ## If `p` is **nil** then a new memory block is returned if alignment `align`.
```

Comment:
Should be 'with alignment'

Reply:
Yep, plus there is a bug in createShared